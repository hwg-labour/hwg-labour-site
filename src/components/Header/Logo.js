import { bpEither } from "codogo-utility-functions";

import Link from "gatsby-link";
import logo from "../../images/logo.png";
import React from "react";
import styled from "styled-components";

const LogoWrapper = styled(Link)`
  ${bpEither("left", "30px")};
  display: flex;
  flex-direction: row;
  align-items: center;
  width: 100%;
  height: 75%;
  padding: 0.75em 0;
`;

const LogoImage = styled.img`
  width: 100%;
  height: 100%;
  max-width: 100%;
  max-height: 100%;
  object-fit: contain;
  object-position: left;
`;

const Logo = () => (
  <LogoWrapper to="/">
    <LogoImage src={logo} />
  </LogoWrapper>
);

export default Logo;
