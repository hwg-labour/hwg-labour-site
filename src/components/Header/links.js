import slugify from "slugify";

const getInvolvedDropdownLinks = [
  {
    to: "/new-members/",
    content: "New Members",
    as: "gatsby-link",
  },
  {
    to: "/whats-on/",
    content: "What's On",
    as: "gatsby-link",
  },
  {
    to: "/campaigning/",
    content: "Campaigning",
    as: "gatsby-link",
  },
  {
    to: "/volunteering/",
    content: "Volunteering",
    as: "gatsby-link",
  },
  {
    to: "/contact-us/",
    content: "Contact Us",
    as: "gatsby-link",
  },
  {
    to: "https://donation.labour.org.uk/page/contribute/donate-fa/",
    content: "Donate",
    as: "a",
  },
  {
    to: "https://join.labour.org.uk/",
    content: "Join",
    as: "a",
  },
];

const peopleDropdownLinks = [
  {
    to: "/mp/",
    content: "Catherine West MP",
    as: "gatsby-link",
  },
  {
    to: "/am/",
    content: "Joanne McCartney AM",
    as: "gatsby-link",
  },
  {
    to: "/mayor/",
    content: "Sadiq Khan",
    as: "gatsby-link",
  },
  {
    to: "/councillors/",
    content: "Councillors",
    as: "gatsby-link",
  },
];

const getGroupsDropdownLinks = ({ edges, }) =>
  edges
    .sort((x, y) => {
      return x.node.name.toUpperCase() < y.node.name.toUpperCase()
        ? -1
        : 1;
    })
    .map(group => {
      return {
        to: `/groups/${ slugify(group.node.name, {
          lower: true,
        }) }`,
        content: group.node.name,
        as: "gatsby-link",
      };
    });

const getWardsDropdownLinks = ({ edges, }) =>
  edges
    .sort((x, y) => {
      return x.node.name.toUpperCase() < y.node.name.toUpperCase()
        ? -1
        : 1;
    })
    .map(ward => {
      return {
        to: `/wards/${ slugify(ward.node.name, {
          lower: true,
        }) }`,
        content: ward.node.name,
        as: "gatsby-link",
      };
    });

export const getLinks = (wards, groups) => {
  return [
    {
      as: "gatsby-link",
      to: "/new-members/",
      content: "Get Involved",
      dropdown: getInvolvedDropdownLinks,
    },
    {
      as: "gatsby-link",
      to: "/wards/",
      content: "Wards",
      dropdown: [
        ...getWardsDropdownLinks(wards),
        {
          to:
            "http://www.haringey.gov.uk/local-democracy/councillors-and-mps/find-my-ward",
          content: "Find your ward",
          as: "a",
        },
      ],
    },
    {
      as: "gatsby-link",
      to: "/groups/",
      content: "Forums & Groups",
      dropdown: getGroupsDropdownLinks(groups),
    },
    {
      as: "gatsby-link",
      to: "/councillors/",
      content: "People",
      dropdown: peopleDropdownLinks,
    },
    {
      as: "gatsby-link",
      to: "/news/",
      content: "News",
    },
    {
      as: "a",
      to: "https://medium.com/hornsey-and-wood-green-labour",
      content: "Thoughts",
    },
    {
      as: "a",
      to: "https://donate.labour.org.uk/hornsey_wood_green/1",
      content: "Donate",
      border: true,
    },
    {
      as: "a",
      to: "https://join.labour.org.uk/",
      content: "Join",
      border: true,
    },
  ] 
}