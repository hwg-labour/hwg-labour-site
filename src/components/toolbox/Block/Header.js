import React, { Component } from 'react'

export default class Header extends Component {
  static className = 'BlockHeader'

  render() {
    const { children, className, ...rest } = this.props

    return (
      <div {...rest}>
        {children}
      </div>
    )
  }
}
