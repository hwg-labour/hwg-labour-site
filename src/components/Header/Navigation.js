import { Nav } from '../toolbox';

import Logo from "./Logo";
import PropTypes from "prop-types";
import React from "react";

import {
	getLinks,
} from './links'

const Navigation = ( { homepage, wards, groups }, ) => (
	<Nav
		logo = { <Logo /> }
		clear = { homepage ? true : false }
		fixed = { homepage ? false : true }
		links = { getLinks(wards, groups) }
	/>
);

Navigation.propTypes = {
	homepage: PropTypes.bool,
	wards: PropTypes.object,
	groups: PropTypes.object,
};

export default Navigation;
