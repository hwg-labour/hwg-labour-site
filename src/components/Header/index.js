import React from "react";
import PropTypes from "prop-types";

import Header from "./Header";

import Banner from "../Banner";

// ----------------------------------------------------

const Header = props => (
	<div>
		<Header { ...props } />

		{props.homepage && <Banner />}

		{props.children}
	</div>
);

Header.propTypes = {
	children: PropTypes.node,
};

// ----------------------------------------------------

export default Header;
