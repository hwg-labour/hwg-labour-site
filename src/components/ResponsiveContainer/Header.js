import React from "react";
import Link from "gatsby-link";
import slugify from "slugify";

import { Nav, } from "codogo-nav";

import Logo from "./Logo";

import {
	Button,
} from "semantic-ui-react";

// ----------------------------------------------------

const GetInvolvedDropdown = () => (<div>
	<Link
		to = "/new-members/"
		className = "ui"
	>
		New Members
	</Link>

	<Link
		to = "/whats-on/"
		className = "ui"
	>
		What's On
	</Link>

	<Link
		to = "/campaigning/"
		className = "ui"
	>
		Campaigning
	</Link>

	<Link
		to = "/volunteering/"
		className = "ui"
	>
		Volunteering
	</Link>

	<a
		href = "https://donation.labour.org.uk/page/contribute/donate-fa/"
		className = "ui"
	>
		Donate
	</a>
</div>);

const PeopleDropdown = () => (
	<div>
		<Link to = "/mp/" className = "ui">
			Catherine West MP
		</Link>

		<Link to = "/am/" className = "ui">
			Joanne McCartney AM
		</Link>

		<Link to = "/mayor/" className = "ui">
			Your Mayor
		</Link>

		<Link
			to = "/councillors/"
			className = "ui"
		>
			Councillor Candidates
		</Link>
	</div>
);

const GroupsDropdown = () => (
	<div>
		{
			props.wards.edges.sort( (x, y) => { return x.node.name.toUpperCase() < y.node.name.toUpperCase() ? -1 : 1 } ).map(ward => (
				<Link
					to = { `/wards/${ slugify(
						ward.node.name,
						{ lower: true, },
					) }` }
					className = "ui"
					key = { ward.node.id + "-dropdown" }
				>
					{ ward.node.name }
				</Link>
			))
		}

		<a
			href = "http://www.haringey.gov.uk/local-democracy/councillors-and-mps/find-my-ward"
			className = "ui"
		>
			Find your ward
		</a>
	</div>
);

// ----------------------------------------------------

const Header = props => (
	<Nav
		logo = { <Logo /> }
		fontSize = { { xs: "1.1em", other: "0.8em", } }
		textTransform = { { xs: "capitalize", other: "capitalize", } }
		color = { { xs: "white", other: "white", } }
		clear
	>
		<Link
			to = "/new-members/"
			className = "ui"
			data-dropdown = { GetInvolvedDropdown }
		>
			Get Involved
		</Link>

		<Link
			to = "/new-members/"
			className = "ui"
			data-dropdown = { GroupsDropdown }
		>
			Groups
		</Link>
		
		<Link
			to = "/people/"
			className = "ui"
			data-dropdown = { PeopleDropdown }
		>
			People
		</Link>

		<Link
			to = "/news/"
			children = "News"
		/>

		<a
			href = "https://medium.com/hornsey-and-wood-green-labour"
			children = "Thoughts"
		/>

		<Link
			to = "/contact-us/"
			children = "Contact Us"
		/>

		<Button
			as = "a"
			inverted = { true }
			style = { { marginLeft: "0.5em", } }
			href = "https://join.labour.org.uk/"
		>
			Donate
		</Button>

		<Button
			as = "a"
			inverted = { true }
			style = { { marginLeft: "0.5em", } }
			href = "https://join.labour.org.uk/"
		>
			Join
		</Button>
	</Nav>
);

export default Header;