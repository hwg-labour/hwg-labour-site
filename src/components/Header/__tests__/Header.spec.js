import Header from "../index";
import React from "react";
import { shallow } from "enzyme";

describe("Header", () => {
  it("renders component", () => {
    const wrapper = shallow(<Header />);

    expect(wrapper).toMatchSnapshot();
  });
});
