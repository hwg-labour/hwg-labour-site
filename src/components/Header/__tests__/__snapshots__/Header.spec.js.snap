// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders component 1`] = `
<div>
  <Navigation />
</div>
`;
